app/screens/channel/channel_nav_bar/channel_title/index.js

```diff
@@ -3,22 +3,40 @@
 
 import {connect} from 'react-redux';
 
-import {getCurrentChannel, getMyCurrentChannelMembership} from 'mattermost-redux/selectors/entities/channels';
+import {General} from 'mattermost-redux/constants';
+import {getCurrentChannel, getMyCurrentChannelMembership, getCurrentChannelStats} from 'mattermost-redux/selectors/entities/channels';
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
-import {isChannelMuted} from 'mattermost-redux/utils/channel_utils';
+import {getCurrentUserId, getUser} from 'mattermost-redux/selectors/entities/users';
+import {getUserIdFromChannelName, isChannelMuted} from 'mattermost-redux/utils/channel_utils';
+
+import {isGuest} from 'app/utils/users';
 
 import ChannelTitle from './channel_title';
 
 function mapStateToProps(state) {
     const currentChannel = getCurrentChannel(state);
+    const currentUserId = getCurrentUserId(state);
     const myChannelMember = getMyCurrentChannelMembership(state);
+    const stats = getCurrentChannelStats(state) || {member_count: 0, guest_count: 0};
+
+    let isTeammateGuest = false;
+    if (currentChannel && currentChannel.type === General.DM_CHANNEL) {
+        const teammateId = getUserIdFromChannelName(currentUserId, currentChannel.name);
+        const teammate = getUser(state, teammateId);
+        if (isGuest(teammate)) {
+            isTeammateGuest = true;
+        }
+    }
 
     return {
         currentChannelName: currentChannel ? currentChannel.display_name : '',
         isArchived: currentChannel ? currentChannel.delete_at !== 0 : false,
         displayName: state.views.channel.displayName,
+        channelType: currentChannel.type,
         isChannelMuted: isChannelMuted(myChannelMember),
         theme: getTheme(state),
+        isGuest: isTeammateGuest,
+        hasGuests: stats.guest_count > 0,
```

Comment:
is this backwards compatible?

Reply:
I think so, if you don't have the guest_count, it will be undefined and `undefined > 0` is false, which is the expected value for "hasGuests" in a old server.